websocket/client.nim

```diff
@@ -43,7 +43,7 @@ proc newAsyncWebsocket*(host: string, port: Port, path: string, ssl = false,
     additionalHeaders: seq[(string, string)] = @[],
     protocols: seq[string] = @[],
     userAgent: string = WebsocketUserAgent,
-    ctx: SslContext = newContext(protTLSv1)
+    ctx: SslContext = when defined(ssl): newContext(protTLSv1) else: nil
```

Comment:
Is ``SslContext`` defined without -d:ssl?

Reply:
It's there now, but is that going to stick around?

https://github.com/nim-lang/Nim/commit/1fb5dd247745e06c85a2ed1ad8f8f09f0a38b9b9

Reply:
Hrm, yeah, I think so.

I think it would make sense to introduce a ``defaultSslContext`` variable (like the one used in the httpclient module) in the future. But this code should continue to work.